packages/crayons-core/src/components/popover/popover.tsx

```diff
@@ -239,6 +248,32 @@ export class Popover {
     };
   }
 
+  handlePopOverShow() {
+    clearTimeout(this.timerId);
+    if (this.showAfter > 0) {
+      this.showWithDelay();
+    } else {
+      this.show();
+    }
+  }
+
+  handlePopOverHide() {
+    clearTimeout(this.timerId);
+    if (this.hideAfter > 0) {
+      this.hideWithDelay();
+    } else {
+      this.hide();
+    }
+  }
+
+  async showWithDelay() {
```

Comment:
Can you please explain why this and the below methods are marked as async ?
Marking something as async will create a promise for us, which is not required for us in this scenario.

Reply:
This async is not needed anymore, removed it. Got an error insisting to add this earlier while playing around, now looks fine without it. Thanks for pointing out.